R/utils.R

```diff
@@ -19,50 +19,14 @@
 #' @return The [`data frame`] object with results from `FUN`.
 #'
 #' @export
+#' @example examples/utils-dapply.R
 #'
-# dapply <- function(df, f, FUN, ...) {
-#   assert_data_frame(df)
-# 
-#   list_df <- split(df, f = f)
-#   list_df <- lapply(list_df, FUN, ...)
-#   do.call(rbind, c(list_df, make.row.names = FALSE))
-# }
+dapply <- function(df, f, FUN, ...) {
+  assert_data_frame(df)
 
-#' Wrapper of tapply, it assumess that the `FUN` returns an output of length
-#' equal to 1 or equal to length of each cell of X. The groupped output of `FUN` is 
-#' not sorted with respect to unique values of `INDEX` but it preserve its ordering
-#' as originally given in `INDEX`. The length of output object is always equal to 
-#' the length of `X`
-#' 
-#' x <- c(0.1, 5, 5, 5, 0.1, 20, 20)
-#' i <- c("B", "B", "B", "B", "A", "A", "A")
-#' 
-#' tapply(x, i, max)
-#' h_tapply(x, i, max)
-#' 
-#' tapply(x, i, `+`, 100)
-#' h_tapply(x, i, `+`, 100)
-#' 
-#' @export
-h_tapply <- function(X, INDEX, FUN, ...) {
-  list_splitted <-  tapply(X, INDEX, FUN, ..., simplify = FALSE)
-  for(i in names(list_splitted)) {
-    X[i == INDEX] <- list_splitted[[i]]
-  }
-  X
-}
-
-#' This is to blind the [`Data`].
-#' For each cohort, the placebo is set to the active dose level for that cohort.
-#' In addition, all DLTs are assigned to the first subjects in the cohort.
-#'
-#' @export
-h_data_blind <- function(x) {
-  assert_class(x, "Data")
-  
-  if (x@placebo) {
-    x@x <- h_tapply(x@x, x@cohort, max)
-    x@y <- h_tapply(x@y, x@cohort, sort, decreasing = TRUE)
-  }
-  x
+  list_df <- split(df, f = f)
+  list_df <- lapply(list_df, FUN, ...)
+  df2 <- do.call(rbind, list_df)
```

Comment:
I think it is a bit better to define the arg list for rbind together with the make row names = FALSE to not get any rownames in the first place

Reply:
Yap, this is how it was, but then, depending on the output class from `FUN`, the `make row names = FALSE` is treated as a new record to `rbind`, rather than the parameter (for `rbind.data.frame`). See the first example for this function I added to the help page.

Reply:
@wwojciech ok thx. then leave that as is.